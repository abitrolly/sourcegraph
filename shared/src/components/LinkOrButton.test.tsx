import React from 'react'
import { LinkOrButton } from './LinkOrButton'
import { mount } from 'enzyme'

describe('LinkOrButton', () => {
    test('render a link when "to" is set', () => {
        expect(mount(<LinkOrButton to="http://example.com">foo</LinkOrButton>)).toMatchSnapshot()
    })

    test('render a button when "to" is undefined', () => {
        expect(mount(<LinkOrButton to={undefined}>foo</LinkOrButton>)).toMatchSnapshot()
    })
})
