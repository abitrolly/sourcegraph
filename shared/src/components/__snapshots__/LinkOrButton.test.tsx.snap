// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkOrButton render a button when "to" is undefined 1`] = `
<LinkOrButton>
  <a
    className="nav-link"
    href=""
    onAuxClick={[Function]}
    onClick={[Function]}
    onKeyPress={[Function]}
    tabIndex={0}
  >
    foo
  </a>
</LinkOrButton>
`;

exports[`LinkOrButton render a link when "to" is set 1`] = `
<LinkOrButton
  to="http://example.com"
>
  <AnchorLink
    className="nav-link"
    onClick={[Function]}
    onKeyPress={[Function]}
    tabIndex={0}
    to="http://example.com"
  >
    <a
      className="nav-link"
      href="http://example.com"
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={0}
    >
      foo
    </a>
  </AnchorLink>
</LinkOrButton>
`;
